src/balloonpanel/balloonpanelview.js

```diff
@@ -67,6 +71,40 @@ export default class BalloonPanelView extends View {
 		this.set( 'isVisible', false );
 
 		/**
+		 * Positions taken intro consideration when attaching the panel
+		 * using {@link ui.balloonPanel.BalloonPanelView#attachTo} method.
+		 * Corresponding with {@link ui.balloonPanel.BalloonPanelView#defaultPositions}.
+		 *
+		 * See {@link ui.balloonPanel.BalloonPanelView#defaultPositions} to learn
+		 * about default available positions.
+		 *
+		 * Also see {@link ui.balloonPanel.BalloonPanelView#position}.
+		 *
+		 * @type {Array.<String>}
+		 */
+		this.positions = [
```

Comment:
Prototyping contextual toolbar I already learned that this is not the best way of configuration.

1. First of all, a contextual toolbar must bring 3 new positions to `BalloonPanelView.defaultPositions` without affecting the prototype of the entire class. `this.positions` should rather contain actual functions than strings or `BalloonPanelView.defaultPositions` should become `this.defaultPositions`.
2. Secondly, it turned out that positions may change a lot with each `attachTo`, like this
```js
editor.listenTo( viewDocument, 'click', () => {
	if ( viewDocument.selection.isCollapsed ) {
		balloonPanelView.positions = [ 's' ];
	} else {
		balloonPanelView.positions =  [
			viewDocument.selection.isBackward ? 'backwardSelection' : 'forwardSelection'
		];
	}

	balloonPanelView.attachTo( viewDocument.domConverter.viewRangeToDom( viewDocument.selection.getFirstRange() ) );
} );
```
  so probably `attachTo` should accept options object which would be merged with some default options, like
```js
balloonPanelView.attachTo( {
	target: viewDocument.domConverter.viewRangeToDom( viewDocument.selection.getFirstRange() ),
	positions: [
		viewDocument.selection.isBackward ? 'backwardSelection' : 'forwardSelection'
	]
} );
```
mimicking `getOptimalPosition` API.